src/components/UserInputComponent/WayareaComponent.js

```diff
@@ -29,51 +54,126 @@ export default function WayareaNameComponent(props) {
       });
   }, []);
 
+  const handleChange = (event) => {
+    SetCalculationInterval(event.target.value);
+    formik.setFieldValue("calculation_interval", event.target.value);
+  };
+  const MenuOptions = () => {
+    return calculationIntervalOptions.map((e) => (
+      <MenuItem key={e} value={e}>
+        {e}
+      </MenuItem>
+    ));
+  };
+  const handleTooltipClose = () => {
+    setOpen(false);
+  };
+  const handleTooltipOpen = () => {
+    setOpen((r) => !r);
```

Comment:
```suggestion
    setOpen((prevValue) => !prevValue);
```
I know I told you to just write `setOpen((r) => !r)`, but it could be changed to be clearer :D